src/ui/components/Transcript/Transcript.tsx

```diff
@@ -7,27 +7,40 @@ import { Comment } from "ui/state/comments";
 import CommentCard from "ui/components/Comments/TranscriptComments/CommentCard";
 import useAuth0 from "ui/utils/useAuth0";
 import MaterialIcon from "ui/components/shared/MaterialIcon";
-import { commentKeys } from "ui/utils/comments";
+import { commentKey } from "ui/utils/comments";
+import uniqBy from "lodash/uniqBy";
 
 export default function Transcript() {
   const recordingId = hooks.useGetRecordingId();
-  const { comments } = hooks.useGetComments(recordingId);
-  const { loading } = hooks.useGetRecording(recordingId);
+  const { comments, loading } = hooks.useGetComments(recordingId);
+  const recording = hooks.useGetRecording(recordingId);
+  const auth = useAuth0();
+
   const pendingComment = useSelector(selectors.getPendingComment);
   const { isAuthenticated } = useAuth0();
 
-  if (loading) {
-    return null;
-  }
+  const displayedComments = useMemo(() => {
+    const displayedComments: Comment[] = [...comments].filter(
+      comment => !pendingComment || commentKey(comment) != commentKey(pendingComment.comment)
+    );
 
-  const displayedComments: Comment[] = [...comments];
+    if (pendingComment?.type == "new_comment") {
+      displayedComments.push(pendingComment.comment);
+    }
 
-  if (pendingComment?.type == "new_comment") {
-    displayedComments.push(pendingComment.comment);
-  }
+    const sortedComments = sortBy(displayedComments, ["time", "createdAt"]);
 
-  const sortedComments = sortBy(displayedComments, ["time", "createdAt"]);
-  const keys = commentKeys(sortedComments);
+    if (uniqBy(sortedComments, commentKey).length != sortedComments.length) {
+      console.warn("Duplicate comments detected");
+      debugger;
```

Comment:
this check helped me see where there were dupes so i kept it in to share the approach. It's safe to remove now